vyper/math/log.vy

```diff
@@ -0,0 +1,74 @@
+TEN_POWERS_TABLE: public(map(decimal, decimal))
+
+
+@public
+def __init__():
+    self.TEN_POWERS_TABLE[0.0] = 1.0
+    self.TEN_POWERS_TABLE[1.0] = 10.0
+    self.TEN_POWERS_TABLE[2.0] = 100.0
+    self.TEN_POWERS_TABLE[3.0] = 1000.0
+    self.TEN_POWERS_TABLE[4.0] = 10000.0
+    self.TEN_POWERS_TABLE[5.0] = 100000.0
+    self.TEN_POWERS_TABLE[6.0] = 1000000.0
+    self.TEN_POWERS_TABLE[7.0] = 10000000.0
+    self.TEN_POWERS_TABLE[8.0] = 100000000.0
+    self.TEN_POWERS_TABLE[9.0] = 1000000000.0
+    self.TEN_POWERS_TABLE[10.0] = 10000000000.0
+    self.TEN_POWERS_TABLE[11.0] = 100000000000.0
```

Comment:
To save gas on SLOADs, you can keep TEN_POWERS_TABLE as a constant() list.
And then when you require it just do `powers: list[decimal] = self.TEN_POWERS_TABLE` This will save a ton of gas :) as you'll only be doing mloads.

Reply:
@bakasura980 - I definitely think we should make this change before merging if you get a moment... And I'm with Jacques, it would be great to use hypothesis as a test framework too...

https://hypothesis.works

Reply:
@andytudhope. I will do the changes. The reason i did not do it yet is because yesterday I was trying to solve the rounding in the power function. If you want, I can resolve the Jacques comments and continuing with the power function after that ?

Reply:
Go in whichever order you prefer @bakasura980, don't wanna get in the way of you solving the actual power function ❤️ No big rush here.

Reply:
This is super. Thank you @andytudhope !

Reply:
@jacqueswww, `powers: decima[12] = TEN_POWERS_TABLE` returns me ''ListType' object has no attribute 'typ'' . I am up to date with the master branch

Reply:
Strange, will have a look to fix that.